app_backend/server.py

```diff
@@ -85,18 +86,45 @@ class User(BaseModel):
 
 
 """
-    getuser implements the route /login/
-    returns message
+    login implements the route /login/
+
+    login: User
+        The UserLogin model that includes the username, email, and password
+    resp: Response
+        The response to send back to the user which contains the status code and body
+    returns Response
+        Response.body: dict
+            Provides any msgs/errs for the request
+        Response.status: int
+            The status code for the request
 """
-# TODO: Once attached to database will be able to retreive
-# specific user information and to log them in
-@app.get("/login")
-async def login():
-    return {"msg": "This is the login page!"}
+@app.post("/login")
+async def login(login: User, resp: Response):
+    name = login.name
+    email = login.email
+    password = login.password
 
+    user = {
+        "name": name,
+        "email": email,
+        "password": password
+    }
+
+    db_user = db.get_user_by_email(user)
+
+    if db_user is None:
+        resp.status_code = status.HTTP_400_BAD_REQUEST
+        return {"err": "This user does not exist. Please try different credentials."}
+    
+    if password != db_user['password']:
+        resp.status_code = status.HTTP_400_BAD_REQUEST
+        return {"err": "Password does not match. Please try a different password."}
+        
+    return {"msg": "Successfully logged in"}
```

Comment:
This is fine, but ultimately presents a security risk. We don't want to tell hackers exactly what exists and what doesn't. Instead, modify `get_user_by_email` to become, `get_user` which tries to find a user with the provided email and password. If our results is 0, then we say that either the email or password is incorrect. Otherwise, this user is authenticated and we log them in.